pkgs/pkgs-lib/formats.nix

```diff
@@ -517,4 +517,37 @@ rec {
     '') {};
   };
 
+  xml = {}: json {} // {
```

Comment:
Is this format suitable for markup-like XML, such as XHTML? Based on the examples here and [there](https://omz-software.com/pythonista/docs/ios/xmltodict.html), that seems to be an unsolved problem.

I doubt that this is the only and best Nix representation of XML, so I'd prefer a more specific name, like `xmltodict`.

Reply:
it is the other way around: `attrsToXml` right?

maybe we include the `badgerfish` notation in the name? like `badgerfishToXml`?

Reply:
Only referred to `xmltodict` as the project name, assuming they didn't name the input format. So they implement the badgerfish format? Then that's great!

I guess we could group representations of XML under an `xml` attrset, so that it's `formats.xml.badgerfish`? That way it's more easily discoverable in documentation and the repl.

Reply:
Or `formats.xml.fromBadgerfish`.
`formats.badgerFishToXml` also works. We don't have nesting yet, actually.

Maybe @infinisil has an opinion, as he's started this library.
Otherwise, current name will do.

Reply:
I am open for suggestions. `formats.xml.fromBadgerfish` sounds good. but lets wait for feedback from others.

Reply:
What about `formats.xml { format = "badgerfish" }` (probably the default value)? This seems to be the intent with having the attrset parameter.

Reply:
implemented the idea from @ambroisie

Reply:
ping @infinisil

Reply:
i someone is looking for different kinds of formats here is a good list: https://wiki.open311.org/JSON_and_XML_Conversion/

But when searching for implementations on GitHub;  `xmltodict` is the most actively maintained and famous one.

So IMHO it make sense to have this as default, because it is the only thing that has "prevailed"